src/cache_handler.rs

```diff
@@ -58,7 +54,7 @@ impl Handler for StaticWithCache {
         match requested_path.get_file() {
             Some(file) => {
                 let last_modified_time = match file.stat() {
-                    Err(error) => return Err(box FileError(error) as IronError),
+                    Err(error) => return Err(Box::new(error)),
```

Comment:
The story here is complex and shifting, particularly because I am dissatisfied with the current state of affairs regarding errors and am exploring alternatives. My primary candidate right now is basically a wrapper type around (error, modifier) that removes most of the "catching" and handling logic of the current system in favor of what's basically a short-circuit.

Within the current system this is basically the right thing to do - though a wrapper type that gives more details (what action were we attempting when this error occurred?) is probably better.